.github/workflows/upload_benchmark_results.yml

```diff
@@ -0,0 +1,104 @@
+name: Upload SDGym Benchmark results
+
+on:
+  push:
+    branches:
+      - issue-425-workflow-sdgym
+  workflow_run:
+    workflows: ["Run SDGym Benchmark"]
+    types:
+      - completed
+  workflow_dispatch:
+  schedule:
+    - cron: '0 6 * * *'
+
+jobs:
+  upload-sdgym-benchmark:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up latest Python
+      uses: actions/setup-python@v5
+      with:
+        python-version-file: 'pyproject.toml'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -e .[dev]
+          
+    - name: Upload SDGym Benchmark
+      env:
+        PYDRIVE_TOKEN: ${{ secrets.PYDRIVE_TOKEN }}
+        AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        GITHUB_LOCAL_RESULTS_DIR: ${{ runner.temp }}/sdgym-leaderboard-files
+      run: |
+        invoke upload-benchmark-results
+        echo "GITHUB_LOCAL_RESULTS_DIR=$GITHUB_LOCAL_RESULTS_DIR" >> $GITHUB_ENV
+
+    - name: Prepare files for commit
+      if: env.SKIP_UPLOAD != 'true'
+      run: |
+        mkdir pr-staging
+        echo "Looking for files in: $GITHUB_LOCAL_RESULTS_DIR"
+        ls -l "$GITHUB_LOCAL_RESULTS_DIR"
+        for f in "$GITHUB_LOCAL_RESULTS_DIR"/*; do
+          if [ -f "$f" ]; then
+            base=$(basename "$f")
+            cp "$f" "pr-staging/${base}"
+          fi
+        done
+
+        echo "Files staged for PR:"
+        ls -l pr-staging
+
+    - name: Checkout target repo (sdv-dev.github.io)
+      if: env.SKIP_UPLOAD != 'true'
+      run: |
+        git clone https://github.com/sdv-dev/sdv-dev.github.io.git target-repo
+        cd target-repo
+        git checkout gatsby-home
```

Comment:
why don't we use the [github checkout action](https://github.com/actions/checkout) for this?